flang/test/Lower/global-init.f90

```diff
@@ -0,0 +1,23 @@
+! RUN: bbc %s -o - | FileCheck %s
+
+program bar
+! CHECK: {{.*}}fir.address_of(@_QEmy_data)
+! CHECK: fir.global {{.*}}Emy_data
+  integer, save :: my_data = 1
+  print *, my_data
+call foo
+contains
+subroutine foo()
+! CHECK: {{.*}}fir.address_of(@_QFfooEmy_data)
+! CHECK: fir.global {{.*}}fooEmy_data
```

Comment:
Why not make sure the `address_of` symbol matches the `global` symbol exactly? You can use the matching REs to do that and make it robust against mangling changes.

Reply:
Like so?
```suggestion
! CHECK: fir.address_of({{.*}}foo{{.*}}my_data)
! CHECK: fir.global {{.*}}foo{{.*}}my_data
```